### PR TITLE
Add more arithmetic bloqs.

### PR DESCRIPTION
**Pull request by fdmalone**
* Square real
* Scale int by real
* Multiply two reals

Fix bug with bitsize for sum of squares.

### CODE REVIEW

**Comment by fdmalone** [comment]
Pulled out from #359

**Comment by fdmalone** [comment]
Oh and an out of place adder which raised #375

File: qualtran/bloqs/arithmetic.py [hunk 1]
```diff
@@ -50,12 +58,58 @@ def t_complexity(self):
...
+
+    @property
+    def signature(self):
+        return Signature.build(a=self.bitsize, b=self.bitsize, c=self.bitsize)
+
+    def pretty_name(self) -> str:
+        return "c = a + b"
+
+    def t_complexity(self):
+        # TODO: This is just copied from adder above, not sure if correct.
```
Comment (NoureldinYosri) [inline qualtran/bloqs/arithmetic.py:96]: T count is correct. Clifford count should be inceased by `bitsize`.

the decomposition would be

```py3
yield CNOT.on_each(zip(b, c))
yield Add(a, c)
```

Comment (fdmalone) [reply qualtran/bloqs/arithmetic.py:96]: Awesome thanks!

Comment (mpharrigan) [reply qualtran/bloqs/arithmetic.py:96]: can we capture this somehow? i.e. by using bloq_counts of `n` cnots and an `Add`

**Review by mpharrigan — APPROVED** [review]
giving approval, but try to incorporate nour's comment

File: qualtran/bloqs/arithmetic.py (hunk 1 above)
Comment (fdmalone) [reply qualtran/bloqs/arithmetic.py]: Done

**Pull request merged**